apps/server/src/modules/files/repo/sync-files.repo.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { cleanupCollections, fileFactory, fileRecordFactory, taskFactory } from '@shared/testing';
+import { SyncFilesRepo } from './sync-files.repo';
+
+// This repo is used for syncing the new filerecords collection with the old files collection.
+// It can be removed after transitioning file-handling to the new files-storage-microservice is completed.
+
+describe('SyncTaskRepo', () => {
+	let module: TestingModule;
+	let repo: SyncFilesRepo;
+	let em: EntityManager;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			imports: [MongoMemoryDatabaseModule.forRoot()],
+			providers: [SyncFilesRepo],
+		}).compile();
+
+		repo = module.get(SyncFilesRepo);
+		em = module.get(EntityManager);
+	});
+
+	afterEach(async () => {
+		await cleanupCollections(em);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('getTaskFilesToSync', () => {
+		it('should not return tasks without associated files', async () => {
+			const task = taskFactory.build({ name: 'task with file' });
+			await em.persistAndFlush(task);
+
+			const result = await repo.findTaskFilesToSync();
+
+			expect(result).toHaveLength(0);
+		});
+
+		it('should return tasks with files without a corresponding filerecord', async () => {
+			const file = fileFactory.build();
+			await em.persistAndFlush(file);
+			const task = taskFactory.build({ name: 'task with file' });
+			task.files.add(file);
+			await em.persistAndFlush(task);
```

Comment:
const files = fileFactory.buildList(1);
const task = taskFactory.build({ name: 'task with file', files });
await em.persistAndFlush(task);

Should work and do the same, is more readable.